### PR TITLE
[no-release-notes] Add sanity check to `Put`

### PR DESCRIPTION
**Pull request by tbantle22**

### CODE REVIEW

File: go/store/prolly/commit_closure_test.go [hunk 1]
```diff
@@ -122,19 +125,19 @@ func TestCommitClosure(t *testing.T) {
...
 		assert.NoError(t, err)
-		err = e.Add(ctx, NewCommitClosureKey(ns.Pool(), 2, hash.Parse("00000000000000000000000000000000")))
+		err = e.Add(ctx, NewCommitClosureKey(ns.Pool(), 2, addr))
 		assert.NoError(t, err)
 		ccr, err = e.Flush(ctx)
 		assert.NoError(t, err)
 		ccrc, err := ccr.Count()
 		require.NoError(t, err)
-		assert.Equal(t, 4, ccrc)
+		assert.Equal(t, 3, ccrc) // TODO(taylor): why did this change from 4?
```
Comment (tbantle22) [inline go/store/prolly/commit_closure_test.go:140]: Is it okay that this count changed? Or is something wrong

File: go/store/types/value_store.go [hunk 2]
```diff
@@ -404,7 +418,10 @@ func (lvs *ValueStore) bufferChunk(ctx context.Context, v Value, c chunks.Chunk,
 				return err
 			}
 		}
-		return lvs.cs.Put(ctx, c)
+
+		return lvs.cs.Put(ctx, c, func(ctx context.Context, c chunks.Chunk) (hash.HashSet, error) {
+			return hash.NewHashSet(), nil
+		}) // Using lvs.getAddrs here makes a few store/types tests fail
```
Comment (tbantle22) [inline go/store/types/value_store.go:424]: This was kinda fixed, still one test failing here https://github.com/dolthub/dolt/pull/5096/files#diff-7943a083af8107048b13b1758b33001082c0d206a8135df1b8a7e926e22dc27dL346

File: go/store/prolly/commit_closure_test.go (hunk 1 above)
Comment (andy-wm-arthur) [reply go/store/prolly/commit_closure_test.go:140]: lines 127 and 129 had two different addresses before, now they both have `addr`

File: go/store/valuefile/file_value_store.go [hunk 3]
```diff
@@ -101,7 +102,9 @@ func (f *FileValueStore) WriteValue(ctx context.Context, v types.Value) (types.R
 			return types.Ref{}, err
 		}
 
-		err = f.Put(ctx, c)
+		err = f.Put(ctx, c, func(ctx context.Context, c chunks.Chunk) (hash.HashSet, error) {
+			return nil, nil // TODO (taylor): Is there a way to get referenced addr out of the chunk here?
+		})
```
[the remark this answers is not in the document]
Comment (andy-wm-arthur) [reply go/store/valuefile/file_value_store.go:107]: you'll probably have to expose `decodeFromBytes`:
https://github.com/dolthub/dolt/blob/main/go/store/types/codec.go#L55

File: go/store/types/value_store.go (hunk 2 above)
Comment (andy-wm-arthur) [reply go/store/types/value_store.go:424]: we definitely need to fix this before we land the project, this is the primary write path for most things. We're better off skipping those tests than leaving a noop `GetAddrsCb`

**Review by andy-wm-arthur — APPROVED** [review]
Looks pretty good, a few comments

**Pull request merged**